sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedSource.java

```diff
@@ -74,6 +81,7 @@ public List<KafkaUnboundedSource<K, V>> split(int desiredNumSplits, PipelineOpti
             if (pattern.matcher(entry.getKey()).matches()) {
               for (PartitionInfo p : entry.getValue()) {
                 partitions.add(new TopicPartition(p.topic(), p.partition()));
+                Lineage.getSources().add("kafka", ImmutableList.of(bootStrapServers, p.topic()));
```

Comment:
acknowledged, we can have a helper class in KafkaIO's package and deal with these logics there. Then source/sink will use this Helper class instead of call Lineage.getSources/Sinks directly